File: src/easy/ThirdMax.swift
func thirdMax(_ nums: [Int]) -> Int {
  let unique = Set(nums)
  var first = -(1 << 53) + 1
  var second = first
  var third = first
  for n in unique {
    if first < n {
      third = second
      second = first
      first = n
    } else if second < n {
      third = second
      second = n
    } else if third < n {
      third = n
    }
  }
  if unique.count < 3 {
    return first
  }
  return third
}

File: src/easy/third-max.ts
export const thirdMax = (nums: number[]): number => {
	const unique = new Set(nums)
	let first = Number.MIN_SAFE_INTEGER
	let second = first
	let third = first
	for(const num of unique) {
		if(first < num) {
			third = second
		second = first
		first = num
		} else if(second < num ) {
			third = second
			second = num
		} else if(third < num) {
			third = num
		}
	}
	if(unique.size < 3)  return first;
	return third;
}